tests/test_solver.py

```diff
@@ -229,6 +232,17 @@ def set_analytical_solution(self):
                 A4 = - A1
                 self.u_ana = (A1 + A2 * y + A3 * sinh(y / zeta) +
                               A4 * cosh(y / zeta), 0.)
+            elif self._name_suffix == 'TractionControlled':
+                assert hasattr(self, 'tau0')
+                xi = 1. / (mu * cosh(h / zeta) * zeta ** 2 \
+                           - cosh(h / zeta) ** 2 * (mu * zeta ** 2 - (c1 + c4)) \
+                               + sinh(h / zeta) ** 2 * (mu * zeta ** 2 - (c1 + c4)))
+                A1 = - xi * zeta ** 3 * self.tau0 * sinh(h / zeta)
+                A2 = xi * zeta ** 2 * self.tau0 * cosh(h / zeta)
+                A3 = - A2 * zeta
+                A4 = - A1
+                self.u_ana = (A1 + A2 * y + A3 * sinh(y / zeta) +
+                              A4 * cosh(y / zeta), 0.)
```

Comment:
There is a sign flip in the \xi in eq. (25) of the paper, other than that I got the same result. Also notice that the material parameters are different. This is because the formulation of gradient elasticity is different to ours.

Reply:
Can you add this as a comment in the code?